lib/streamlit/DeltaGenerator.py

```diff
@@ -773,9 +781,20 @@ def bar_chart(self, element, data, width=0, height=0):
             height: 200px
 
         """
-        from streamlit.elements.Chart import Chart
-        chart = Chart(data, type='bar_chart', width=width, height=height)
-        chart.marshall(element.chart)
+        import altair as alt
+
+        if isinstance(data, pd.DataFrame):
+            data = pd.melt(data.reset_index(), id_vars=['index'])
```

Comment:
This means we only reshape `data` when it's of type `dataframe`. But we should support other types of data too, as described in the docstring of this function: pandas.DataFrame, pandas.Styler, numpy.ndarray, Iterable, and dict.

* DataFrame is already being handled
* Style is just a wrapper around DataFrame with some extra style options. I think there's a way to get to the inner DataFrame itself.
* ndarray, iterable, and dict get converted to DataFrame later on in the code

The solution here would be to convert all of those to DataFrame first, with:

```
data = pd.DataFrame(data)
```

...in which case the `if` above is not needed.